api4/config.go

```diff
@@ -108,9 +108,10 @@ func configReload(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func updateConfig(c *Context, w http.ResponseWriter, r *http.Request) {
-	cfg := model.ConfigFromJSON(r.Body)
-	if cfg == nil {
-		c.SetInvalidParam("config")
+	var cfg *model.Config
+	err := json.NewDecoder(r.Body).Decode(&cfg)
+	if err != nil || cfg == nil {
+		c.SetInvalidParamWithErr("config", err)
```

Comment:
What will happen if we have the `cfg == nil` case and pass a nil err? Nothing will panic right?

Same applies for the below cases as well.

Reply:
Nothing happens. It will set the wrapped error to `nil`, just as if no error was wrapped.